public/video-ui/src/components/VideoPage/VideoPage.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import Q from 'q';
+import {fetchComposerId} from '../../services/capi.js';
+
+import {getStore} from '../../util/storeAccessor';
+
+export default class VideoPage extends React.Component {
+
+  state = {
+    composerIds: undefined,
+    pageCreated: false
+  }
+
+  componentDidMount() {
+    if (this.props.usages) {
+      this.getComposerIds(this.props.usages);
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    const oldVideoId = this.props.video && this.props.video.id;
+    const newVideoId = newProps.video && newProps.video.id;
+    const oldUsages = this.props.usages;
+    const newUsages = newProps.usages;
+
+    if (oldVideoId !== newVideoId || oldUsages !== newUsages) {
+      this.getComposerIds(newProps.usages);
+    }
+
+  }
+
+  getComposerUrl = () => {
+    return getStore().getState().config.composerUrl;
+  }
+
+  getComposerIds = (usages) => {
+
+    return Q.all(usages.map(fetchComposerId))
+    .then(ids => {
+      const composerIdsWithUsage = ids.reduce((idsWithUsage, composerId, index) => {
+        if (composerId !== '') {
+          idsWithUsage.push({usage: usages[index], composerId: composerId});
+        }
+        return idsWithUsage;
+      }, []);
+      return this.setState({
+        composerIds: composerIdsWithUsage
+      });
+    })
+  }
+
+  pageCreate = () => {
+
+    this.setState({
+      pageCreated: true
+    });
+
+    const videoPage = {
+      elements: [
```

Comment:
This should probably be extracted out into a helper function (perhaps in a different file) that takes a video and generates the block.